src/adapter/src/optimize.rs

```diff
@@ -284,6 +286,12 @@ impl From<String> for OptimizerError {
 }
 
 impl OptimizerError {
+    pub fn redacted(&self) -> String {
+        match self {
+            _ => format!("{:?}", OptimizerErrorKind::from(self)),
+        }
```

Comment:
isn't this the same as just
```
format!("{:?}", OptimizerErrorKind::from(self))
```
it seems like the `match` is unnecessary?

Reply:
note: same comment for all of the other instances below

Reply:
Yes, but I like this structure because it promotes one-line diffs that allow for deeper errors to be exposed (enums of enums).

Reply:
Ahhh good point, works for me!